Compiled from "CWE586_Explicit_Call_to_Finalize__basic_05.java"
public class testcases.CWE586_Explicit_Call_to_Finalize.CWE586_Explicit_Call_to_Finalize__basic_05 extends testcasesupport.AbstractTestCase {
  private boolean privateTrue;

  private boolean privateFalse;

  public testcases.CWE586_Explicit_Call_to_Finalize.CWE586_Explicit_Call_to_Finalize__basic_05();
    Code:
       0: aload_0
       1: invokespecial #1                  // Method testcasesupport/AbstractTestCase."<init>":()V
       4: aload_0
       5: iconst_1
       6: putfield      #7                  // Field privateTrue:Z
       9: aload_0
      10: iconst_0
      11: putfield      #13                 // Field privateFalse:Z
      14: return

  public void bad() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: getfield      #7                  // Field privateTrue:Z
       4: ifeq          46
       7: new           #16                 // class testcases/CWE586_Explicit_Call_to_Finalize/CWE586_Explicit_Call_to_Finalize__basic_Helper
      10: dup
      11: invokespecial #18                 // Method testcases/CWE586_Explicit_Call_to_Finalize/CWE586_Explicit_Call_to_Finalize__basic_Helper."<init>":()V
      14: astore_1
      15: aload_1
      16: invokevirtual #19                 // Method testcases/CWE586_Explicit_Call_to_Finalize/CWE586_Explicit_Call_to_Finalize__basic_Helper.sayHello:()V
      19: aload_1
      20: invokevirtual #22                 // Method testcases/CWE586_Explicit_Call_to_Finalize/CWE586_Explicit_Call_to_Finalize__basic_Helper.finalize:()V
      23: goto          46
      26: astore_2
      27: ldc           #27                 // String An error occurred.
      29: invokestatic  #29                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      32: aload_1
      33: invokevirtual #22                 // Method testcases/CWE586_Explicit_Call_to_Finalize/CWE586_Explicit_Call_to_Finalize__basic_Helper.finalize:()V
      36: goto          46
      39: astore_3
      40: aload_1
      41: invokevirtual #22                 // Method testcases/CWE586_Explicit_Call_to_Finalize/CWE586_Explicit_Call_to_Finalize__basic_Helper.finalize:()V
      44: aload_3
      45: athrow
      46: return
    Exception table:
       from    to  target type
          15    19    26   Class java/lang/Exception
          15    19    39   any
          26    32    39   any

  private void good1() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: getfield      #13                 // Field privateFalse:Z
       4: ifeq          15
       7: ldc           #35                 // String Benign, fixed string
       9: invokestatic  #29                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      12: goto          48
      15: new           #16                 // class testcases/CWE586_Explicit_Call_to_Finalize/CWE586_Explicit_Call_to_Finalize__basic_Helper
      18: dup
      19: invokespecial #18                 // Method testcases/CWE586_Explicit_Call_to_Finalize/CWE586_Explicit_Call_to_Finalize__basic_Helper."<init>":()V
      22: astore_1
      23: aload_1
      24: invokevirtual #19                 // Method testcases/CWE586_Explicit_Call_to_Finalize/CWE586_Explicit_Call_to_Finalize__basic_Helper.sayHello:()V
      27: aconst_null
      28: astore_1
      29: goto          48
      32: astore_2
      33: ldc           #27                 // String An error occurred.
      35: invokestatic  #29                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      38: aconst_null
      39: astore_1
      40: goto          48
      43: astore_3
      44: aconst_null
      45: astore_1
      46: aload_3
      47: athrow
      48: return
    Exception table:
       from    to  target type
          23    27    32   Class java/lang/Exception
          23    27    43   any
          32    38    43   any

  private void good2() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: getfield      #7                  // Field privateTrue:Z
       4: ifeq          40
       7: new           #16                 // class testcases/CWE586_Explicit_Call_to_Finalize/CWE586_Explicit_Call_to_Finalize__basic_Helper
      10: dup
      11: invokespecial #18                 // Method testcases/CWE586_Explicit_Call_to_Finalize/CWE586_Explicit_Call_to_Finalize__basic_Helper."<init>":()V
      14: astore_1
      15: aload_1
      16: invokevirtual #19                 // Method testcases/CWE586_Explicit_Call_to_Finalize/CWE586_Explicit_Call_to_Finalize__basic_Helper.sayHello:()V
      19: aconst_null
      20: astore_1
      21: goto          40
      24: astore_2
      25: ldc           #27                 // String An error occurred.
      27: invokestatic  #29                 // Method testcasesupport/IO.writeLine:(Ljava/lang/String;)V
      30: aconst_null
      31: astore_1
      32: goto          40
      35: astore_3
      36: aconst_null
      37: astore_1
      38: aload_3
      39: athrow
      40: return
    Exception table:
       from    to  target type
          15    19    24   Class java/lang/Exception
          15    19    35   any
          24    30    35   any

  public void good() throws java.lang.Throwable;
    Code:
       0: aload_0
       1: invokevirtual #37                 // Method good1:()V
       4: aload_0
       5: invokevirtual #40                 // Method good2:()V
       8: return

  public static void main(java.lang.String[]) throws java.lang.ClassNotFoundException, java.lang.InstantiationException, java.lang.IllegalAccessException;
    Code:
       0: aload_0
       1: invokestatic  #43                 // Method mainFromParent:([Ljava/lang/String;)V
       4: return
}
